applications/system/js_app/js_thread.c

```diff
@@ -197,13 +198,95 @@ static void js_require(struct mjs* mjs) {
 static void js_global_to_string(struct mjs* mjs) {
     int base = 10;
     if(mjs_nargs(mjs) > 1) base = mjs_get_int(mjs, mjs_arg(mjs, 1));
-    double num = mjs_get_int(mjs, mjs_arg(mjs, 0));
+    double num = mjs_get_double(mjs, mjs_arg(mjs, 0));
     char tmp_str[] = "-2147483648";
     itoa(num, tmp_str, base);
     mjs_val_t ret = mjs_mk_string(mjs, tmp_str, ~0, true);
     mjs_return(mjs, ret);
 }
 
+static void js_parse_int(struct mjs* mjs) {
+    mjs_val_t arg = mjs_arg(mjs, 0);
+    if(!mjs_is_string(arg)) {
+        mjs_return(mjs, mjs_mk_number(mjs, 0));
+        return;
+    }
+    size_t str_len = 0;
+    const char* str = mjs_get_string(mjs, &arg, &str_len);
+    if((str_len == 0) || (str == NULL)) {
+        mjs_return(mjs, mjs_mk_number(mjs, 0));
+        return;
+    }
+
+    int32_t num = 0;
+    int32_t sign = 1;
+    size_t i = 0;
+
+    if(str[0] == '-') {
+        sign = -1;
+        i = 1;
+    } else if(str[0] == '+') {
+        i = 1;
+    }
+
+    for(; i < str_len; i++) {
+        if(str[i] >= '0' && str[i] <= '9') {
+            num = num * 10 + (str[i] - '0');
+        } else {
+            break;
+        }
+    }
+    num *= sign;
+
+    mjs_return(mjs, mjs_mk_number(mjs, num));
+}
+
+static void js_to_upper_case(struct mjs* mjs) {
+    mjs_val_t arg0 = mjs_arg(mjs, 0);
+
+    size_t str_len;
+    const char* str = NULL;
+    if(mjs_is_string(arg0)) {
+        str = mjs_get_string(mjs, &arg0, &str_len);
+    }
+    if(!str) {
```

Comment:
Same here, use ASS

Reply:
while moving to string class, code had to be moved to /lib/mjs, while ASS is part of js_app specifically, so i could not use it.
i made similar code to what was already in mjs string source code, so it looks off compared to normal flipper C but if you look at the rest of the file it fits in. let me know if this is a problem